### DISCUSSION TITLE
Smooth Scrollbar + Shopify

### DISCUSSION THREAD

**Discussion by martinatthedesk** [idiotWu/smooth-scrollbar#549] [Q&A]
Hey, I have a store on shopify and I need help on how to use the smooth scrollbar in shopify please. I'm just a little bit familiar with coding.

**Comment by sadeghbarati** [comment]
Can you share your shopify website?

Are you familiar with bundlers like `Vite`, or `Webpack`?

**Reply by martinatthedesk** [reply]
Unfortunately not, sorry, not sure if the company will be ok with that. 
I'm not familiar, but I learn fast so I'd appreciate your help very much. The shopify theme's is 'Prestige', if it helps. Thanks in advance.

**Comment by sadeghbarati** [comment]
What I understand from Shopify Theme, it uses `ES Modules` and `Web Components`. [more info about ESM Syntax here](https://nuxt.com/docs/guide/concepts/esm#es-modules) and [here](https://flaviocopes.com/es-modules/)


So you can either use `Web Components` or simple HTML elements like `div`
I don't have knowledge about Custom Web Components so I explain with simple HTML elements

```ts
import Scrollbar from 'smooth-scrollbar'

document.addEventListener('DOMContentLoaded', () => {
    Scrollbar.init(document.getElementById('scrollbarElement'), {})
})

// or

window.addEventListener('load', () => {
    Scrollbar.init(document.getElementById('scrollbarElement'), {})
})
```

If you use Web Component you can use [Custom Element lifecycles](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_custom_elements#using_the_lifecycle_callbacks) like `connectedCallback`

**Reply by martinatthedesk** [reply]
Sorry for sounding maybe dumb, but what exactly should I do? Create a .js file in shopify and paste the code that you sent me? Or create a <script></script> in theme.liquid and paste the code there?

**Reply by sadeghbarati** [reply]
Sorry I explain it so hard

Search `<script type="importmap">` in your project and update your `importmap` base on this link 👇 

[JSPM Playground](https://jspm.org/sandbox#H4sIAAAAAAAAA+1WUU/bMBB+pr/iyB4AqYlbQBsqKQK2igEtY6uqaY9ufDQuSRzZbrsw7b/v4rRhRR1T35Ha2r777rvP7vnkcFeoyBY5QmzT5KwRrgbk4qwBEKZoOUQx1wZt15vZB//Ecw4rbYJno8yNImTVug7JeIpdby5xkSttPYhUZjEjioUUNu4KnMsIfbdogsyklTzxTcQT7LYpQcgqBeFYiQKMLcjupVxPZNZpnULOhZDZhKaVmF3fpwHgZng/gCvMUHOrNFynZXIY8Nx5e0JaGH3rd2izNjcdxiYraDA1eRpIxd6NDu+mP64uHz8NWroYPt0MR+34dvDh7qlgN62P7S99o++LUXLb/3y7aPXvvw4WvYvh09ReFqMLyuL7pSAITaRlbqE82q4nnY6U507tLydmaTReZ2kgk0mVsjEdg1ZJMuaafF4tldcaszztvISenwQnwTGTmcCfBPQc5e9mlcpEKse1TBtY/3ITIFIa/alhaBjXmhfsQav0FT1L/PlRcHQYtNfDSj3NjdTlkWzDSfjnLW6mVOMpRpZxY+Qk24p8LfLfCegibEVL+P9pXiB/9Lc9i1XQRnJrEjl+hc75z9tB+5jI3CJA874uHCqdxuq3/Iasqmcq38bL4k6VmCXoKhugKmoYrsoSyv9/Q2GfEtoFMAbftbQIKZcZVFzULAQ1JNTYBLrF3AAHgzmnm4rwIAmwkDYmo2d05AG3VsvxjHwqAxvjiqVSGbg0jcbOTi0qKPvNPjW+WUotKZig7SVYTi+La7G/V6vcOzhY3ztUn1DIOUjR9Z73s2pRMcpJbDvQbrXmcdWcdhz8hf+wdhOgonHzkBG4Mr+FvYW9hW0dtpyErHw5uIeEe9H8AS2vOdHpCAAA)

---

If you **can't** find any importmap in your project, share at least one of your JS source from Shopify Theme

GL 👋
